src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -120,18 +120,20 @@ private bool IsParameterTypeSame(Parameter? first, Parameter? second)
             return first!.Type.Equals(second!.Type);
         }
 
+        private InputType? GetFirstBodyType(InputOperation operation)
+        {
+            var operationBodyTypes = operation.Responses.Where(r => !r.IsErrorResponse).Select(r => r.BodyType).Distinct().ToArray();
+            return operationBodyTypes.Length != 0 ? operationBodyTypes[0] : null;
+        }
+
         private ReturnTypeChain BuildReturnTypes()
         {
-            var operationBodyTypes = Operation.Responses.Where(r => !r.IsErrorResponse).Select(r => r.BodyType).Distinct().ToArray();
             CSharpType? responseType = null;
-            if (operationBodyTypes.Length != 0)
+            var firstBodyType = GetFirstBodyType(Operation);
```

Comment:
this is quite weird, could we have multiple body types?

Reply:
Yes, emitter passes multiple types to generator and generator only takes the first one.

Reply:
Is it request body or response body?
if it is response body, maybe we need to get the body according to the status code, such as response body for 200.

Reply:
It is response body, and yes we already do that.

Reply:
Aactually I don't change any logic here, I just extract that logic into a method for better usage. If you don't like I could revert that part.

Reply:
well to be clear please add some comments here to emphasize this is taking the first of the "response body type"

Reply:
Then maybe need to change the function name `GetFirstBodyType` to avoid confusion. e.g. GetResponseBody

Reply:
I will revert this change, it is irrelevant to the issue I want to fix.